lib/Sema/CSDiag.cpp

```diff
@@ -6246,6 +6282,35 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
 
   auto fnType = getFuncType(CS->getType(fnExpr));
 
+  // Let's see if this has to do with member vs. property error
+  // because sometimes when there is a member and a property declared
+  // on the nominal type with the same name type-checking function
+  // expression separately from arguments might produce solution for
+  // the property instead of the member.
+  if (!fnType->is<AnyFunctionType>()) {
+    if (auto *UDE = dyn_cast<UnresolvedDotExpr>(callExpr->getFn())) {
+      fnExpr = callExpr->getFn();
+
+      SmallVector<Type, 4> types;
+      CS->TC.getPossibleTypesOfExpressionWithoutApplying(fnExpr, CS->DC, types);
+
+      unsigned numFuncTypes = 0;
+      Type funcType;
+      for (auto type : types) {
+        if (getFuncType(type)->is<AnyFunctionType>()) {
+          funcType = getFuncType(type);
+          numFuncTypes++;
+        }
+      }
```

Comment:
This seems like a good place where `std::find_if` followed by `std::none_of` (with the range after the result of `find_if`) would make the code clearer.